narwhal/node/tests/reconfigure.rs

```diff
@@ -1,379 +0,0 @@
-        transaction: Vec<u8>,
-    ) {
-        let transaction: u64 = bincode::deserialize(&transaction).unwrap();
-        // Change epoch every few certificates. Note that empty certificates are not provided to
-        // this function (they are immediately skipped).
-        let mut epoch = self.committee.lock().unwrap().epoch();
-        if transaction >= epoch && execution_indices.next_certificate_index % 3 == 0 {
-            epoch += 1;
-            {
-                let mut guard = self.committee.lock().unwrap();
-                guard.epoch = epoch;
-            };
-
-            let worker_keypairs = self.worker_keypairs.iter().map(|kp| kp.copy());
-            let worker_ids = 0..self.worker_keypairs.len() as u32;
-            let worker_ids_and_keypairs = worker_ids.zip(worker_keypairs).collect();
-
-            let new_committee = self.committee.lock().unwrap().clone();
-
-            self.tx_reconfigure
-                .send((
-                    self.keypair.copy(),
-                    self.network_keypair.copy(),
-                    new_committee,
-                    worker_ids_and_keypairs,
-                    self.worker_cache.clone(),
-                ))
-                .await
-                .unwrap();
-        }
-
-        let _ = self.tx_output.send(epoch).await;
-    }
-
-    async fn load_execution_indices(&self) -> ExecutionIndices {
-        ExecutionIndices::default()
-    }
-}
-
-async fn run_client(
-    name: PublicKey,
-    worker_cache: SharedWorkerCache,
-    mut rx_reconfigure: Receiver<u64>,
-) {
-    let target = worker_cache
-        .load()
-        .worker(&name, /* id */ &0)
-        .expect("Our key or worker id is not in the worker cache")
-        .transactions;
-    let config = mysten_network::config::Config::new();
-    let channel = config.connect_lazy(&target).unwrap();
-    let mut client = TransactionsClient::new(channel);
-
-    // Make a transaction to submit for ever.
-    let mut tx = TransactionProto {
-        transaction: Bytes::from(0u64.to_be_bytes().to_vec()),
-    };
-
-    // Repeatedly send transactions.
-    let mut interval = interval(Duration::from_millis(50));
-    interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
-    tokio::pin!(interval);
-
-    loop {
-        tokio::select! {
-            // Wait a bit before repeating.
-            _ = interval.tick() => {
-                // Send a transactions.
-                if client.submit_transaction(tx.clone()).await.is_err() {
-                    // The workers are still down.
-                    sleep(Duration::from_millis(100)).await;
-                }
-            },
-
-            // Send transactions on the new epoch.
-            Some(epoch) = rx_reconfigure.recv() => {
-                tx = TransactionProto {
-                    transaction: Bytes::from(epoch.to_le_bytes().to_vec()),
-                };
-            }
-        }
-    }
-}
-
-#[tokio::test]
-async fn restart() {
```

Comment:
Let's keep this test, it is needed as it is testing the reconfiguration approach via the `restart/shutdown` method. I'll rebase the PR as I've made some changes on this test